src/CanvasContainer.tsx

```diff
@@ -215,6 +243,43 @@ export const CanvasContainer: React.FC = ({ children }) => {
     };
   }, [canvasService]);
 
+  /**
+   * Keyboard events.
+   */
+  useEffect(() => {
+    function keydownListener(e: KeyboardEvent) {
+      const target = e.target as HTMLElement;
+      if (isTextInputLikeElement(target)) {
```

Comment:
So attaching the listeners directly to the canvas container? I was going to do this initially, only opted for global since that is being done for the spacebar lock logic.

Reply:
It's often easy to focus `body` or some other element - so hooking into a global thing like `window` seems to be preferable because we won't miss keystrokes. 

Sorry if my question was confusing - the question was about if conceptually those kinds of things are truly global, which includes those being triggered also on activatable elements. It was sort of an introduction to the comment about the range input because I've wanted to establish what kind of filtering logic we need to support.